### ISSUE TITLE
ScanWindow not updating

### ISSUE THREAD

**Issue by Sry-Work** [juliansteenbakker/mobile_scanner#975]
Hello,

First of all, thank you for this great library.

In our App we use the [suggested way](https://github.com/juliansteenbakker/mobile_scanner/discussions/17) of rotating the scanner, while also applying a ScanWindow.
While this works great, we did notice that the scanWindow wouldn't be updated after rotating the device.

We tried to manually update it, but the method to transform the Widget space to Camera Space isn't publicly accessible and we would rather not calculate it ourselves, in case the way the transformation is applied/calculated ever changes.

So my suggestion, would be to:

- Automatically recalculate the used scanWindow inside the _MobileScannerState whenever the Layout constraints changed, or the scanWindow parameter changed.

and / or

- Export the calculateScanWindowRelativeToTextureInPercentage method, so that we can calculate the transformed Size
- Change the updateScanWindow method to transform the window parameter, if the user wishes to.
In the best case this should be toggle able with an optional Method parameter

---
Steps to reproduce:
- Create the MobileScanner widget with a scanWindow
- Update the scanWindow variable

**Comment by navaronbracke** [comment]
Thanks for the report. Currently, the scan window is "updated" using a LayoutBuilder [here](https://github.com/juliansteenbakker/mobile_scanner/blob/ad68fda9a8d8730f8cde94bd2c2719a0a0217854/lib/src/mobile_scanner.dart#L148-L150)

However, as you noted there is a bug where the scan window is not updated. I think this is because the scan window is only updated once (when it was null before and now it is not null)

The bug is in this method https://github.com/juliansteenbakker/mobile_scanner/blob/ad68fda9a8d8730f8cde94bd2c2719a0a0217854/lib/src/mobile_scanner.dart#L108-L122

We should update this method to update the scan window when the new computed window is different from the old one.
Due to the LayoutBuilder, we do trigger the recomputation, but it is not comitted, due to this bug.

One thing I worry about, though, is performance.
If the user's screen is being resized, the layout constraints will change rapidly, thus computing a new scan window, which is then sent over the method channel. Perhaps we can do the following:

1) Fix the bug with the scan window check so that it is updated when it changes
2) For performance reasons, we also introduce a threshold for scan window updates. For example, given a threshold of 20.0, we only send the scan window to the barcode scanner when the width or height of the scan window has changed by at least the threshold

What do you think about this proposal?

**Comment by Sry-Work** [comment]
Thank you for the quick reply. 

In my opinion, your proposal sounds great.

**Comment by navaronbracke** [comment]
I will port this fix to both version 4.0.2 and the next 5.0.0 beta release.

**Comment by navaronbracke** [comment]
@Sry-Work could you test if this patch would be sufficient?
This patch is based off version 4.0.1 but I will be porting this to the beta release too.

```
  mobile_scanner:
    git: https://github.com/navaronbracke/mobile_scanner.git
    ref: scan_window_hotfix
```

**Comment by Sry-Work** [comment]
Damn that was quick.

I did a few Test runs with it, and it worked beautifully.

Thanks a bunch :)

**Comment by navaronbracke** [comment]
I'll do an internal test as well, once i've finished that, I'll publish version 4.0.2 (I want to get some additional things into the next beta, so 5.0.0-beta.2 will come a bit later)

**Comment by navaronbracke** [comment]
@Sry-Work I've just published version `5.0.0-beta.2` which includes this fix. I also fixed an oversight in the previous beta, which enforced a higher Flutter constraint, than what was actually needed. You can use the new version if you have Flutter 3.16.0 or higher (rather than requiring Flutter 3.19.0 in the first beta)
